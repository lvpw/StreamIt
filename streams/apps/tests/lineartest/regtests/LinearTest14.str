/** Test out combining splitjoins. **/
void->void pipeline LinearTest14 {
  add FloatSource();
  add LinearSJ();
  add LinearSJ2();
  add LinearSJ3();
  add NonLinearSJ();
  add FloatSink();
}

void->float stateful filter FloatSource {
  float x;
  init {
    x = 0;
  }
  work push 1 {
    push(x);
    x = x + 1;
  }
}

float->void filter FloatSink {
  work pop 1 {
    println(pop());
  }
}

/** A linear split join to test combination out on. **/
float->float splitjoin LinearSJ {
  split duplicate;
  add FloatFilter1();
  add FloatFilter2();
  join roundrobin;
}

/** A linear split join to test combination out on. **/
float->float splitjoin LinearSJ2 {
  split duplicate;
  add FloatFilter1();
  add FloatFilter2();
  add FloatFilter3();
  join roundrobin;
}
/** A linear split join to test combination out on. **/
float->float splitjoin LinearSJ3 {
  split duplicate;
  add FloatFilter1();
  add FloatFilter2();
  add FloatFilter3();
  join roundrobin(4,4,4);
}


/** A non linear split join to make sure that we are sane **/
float->float splitjoin NonLinearSJ {
  split duplicate;
  add FloatFilter1();
  add FloatFilter2();
  add NonLinearFilter();
  join roundrobin;
}




/** filter that computes [[1 2][3 4]] **/
float->float filter FloatFilter1 {
  work pop 2 push 2 {
    float t1 = pop();
    float t2 = pop();
    push(t2*2 + t1*4);
    push(t2*1 + t1*3);
  }
}
/** filter that computes [[5 6][7 8]] **/
float->float filter FloatFilter2 {
  work pop 2 push 2 {
    float t1 = pop();
    float t2 = pop();
    push(t2*6 + t1*8);
    push(t2*5 + t1*7);
  }
}
/** filter that computes [[9 10][11 12]] **/
float->float filter FloatFilter3 {
  work pop 2 push 2 {
    float t1 = pop();
    float t2 = pop();
    push(t2*10 + t1*12);
    push(t2*9 + t1*11);
  }
}


/** nonlinear filter **/
float->float filter NonLinearFilter {
  work pop 2 push 2 {
    float t1 = pop();
    float t2 = pop();
    push (t1*t2);
    push (t2*t1);
  }
}
