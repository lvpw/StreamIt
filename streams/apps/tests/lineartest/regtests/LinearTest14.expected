FloatFilter1_1.matrix
[[1.0+0.0i 2.0+0.0i]
 [3.0+0.0i 4.0+0.0i]]
FloatFilter1_1.vector
[[0.0+0.0i 0.0+0.0i]]
FloatFilter1_2.matrix
[[1.0+0.0i 2.0+0.0i]
 [3.0+0.0i 4.0+0.0i]]
FloatFilter1_2.vector
[[0.0+0.0i 0.0+0.0i]]
FloatFilter1_3.matrix
[[1.0+0.0i 2.0+0.0i]
 [3.0+0.0i 4.0+0.0i]]
FloatFilter1_3.vector
[[0.0+0.0i 0.0+0.0i]]
FloatFilter1_4.matrix
[[1.0+0.0i 2.0+0.0i]
 [3.0+0.0i 4.0+0.0i]]
FloatFilter1_4.vector
[[0.0+0.0i 0.0+0.0i]]
FloatFilter2_1.matrix
[[5.0+0.0i 6.0+0.0i]
 [7.0+0.0i 8.0+0.0i]]
FloatFilter2_1.vector
[[0.0+0.0i 0.0+0.0i]]
FloatFilter2_2.matrix
[[5.0+0.0i 6.0+0.0i]
 [7.0+0.0i 8.0+0.0i]]
FloatFilter2_2.vector
[[0.0+0.0i 0.0+0.0i]]
FloatFilter2_3.matrix
[[5.0+0.0i 6.0+0.0i]
 [7.0+0.0i 8.0+0.0i]]
FloatFilter2_3.vector
[[0.0+0.0i 0.0+0.0i]]
FloatFilter2_4.matrix
[[5.0+0.0i 6.0+0.0i]
 [7.0+0.0i 8.0+0.0i]]
FloatFilter2_4.vector
[[0.0+0.0i 0.0+0.0i]]
FloatFilter3_1.matrix
[[9.0+0.0i 10.0+0.0i]
 [11.0+0.0i 12.0+0.0i]]
FloatFilter3_1.vector
[[0.0+0.0i 0.0+0.0i]]
FloatFilter3_2.matrix
[[9.0+0.0i 10.0+0.0i]
 [11.0+0.0i 12.0+0.0i]]
FloatFilter3_2.vector
[[0.0+0.0i 0.0+0.0i]]
LinearSJ2_1.matrix
[[9.0+0.0i 5.0+0.0i 1.0+0.0i 10.0+0.0i 6.0+0.0i 2.0+0.0i]
 [11.0+0.0i 7.0+0.0i 3.0+0.0i 12.0+0.0i 8.0+0.0i 4.0+0.0i]]
LinearSJ2_1.vector
[[0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i]]
LinearSJ3_1.matrix
[[9.0+0.0i 10.0+0.0i 0.0+0.0i 0.0+0.0i 5.0+0.0i 6.0+0.0i 0.0+0.0i 0.0+0.0i 1.0+0.0i 2.0+0.0i 0.0+0.0i 0.0+0.0i]
 [11.0+0.0i 12.0+0.0i 0.0+0.0i 0.0+0.0i 7.0+0.0i 8.0+0.0i 0.0+0.0i 0.0+0.0i 3.0+0.0i 4.0+0.0i 0.0+0.0i 0.0+0.0i]
 [0.0+0.0i 0.0+0.0i 9.0+0.0i 10.0+0.0i 0.0+0.0i 0.0+0.0i 5.0+0.0i 6.0+0.0i 0.0+0.0i 0.0+0.0i 1.0+0.0i 2.0+0.0i]
 [0.0+0.0i 0.0+0.0i 11.0+0.0i 12.0+0.0i 0.0+0.0i 0.0+0.0i 7.0+0.0i 8.0+0.0i 0.0+0.0i 0.0+0.0i 3.0+0.0i 4.0+0.0i]]
LinearSJ3_1.vector
[[0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i]]
LinearSJ_1.matrix
[[5.0+0.0i 1.0+0.0i 6.0+0.0i 2.0+0.0i]
 [7.0+0.0i 3.0+0.0i 8.0+0.0i 4.0+0.0i]]
LinearSJ_1.vector
[[0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i]]
LinearWrapper_1.matrix
[[856.0+0.0i 944.0+0.0i 754.0+0.0i 824.0+0.0i 504.0+0.0i 592.0+0.0i 474.0+0.0i 544.0+0.0i 152.0+0.0i 240.0+0.0i 194.0+0.0i 264.0+0.0i 496.0+0.0i 548.0+0.0i 1168.0+0.0i 1288.0+0.0i 288.0+0.0i 340.0+0.0i 688.0+0.0i 808.0+0.0i 80.0+0.0i 132.0+0.0i 208.0+0.0i 328.0+0.0i 1032.0+0.0i 1128.0+0.0i 688.0+0.0i 760.0+0.0i 648.0+0.0i 744.0+0.0i 400.0+0.0i 472.0+0.0i 264.0+0.0i 360.0+0.0i 112.0+0.0i 184.0+0.0i]
 [1480.0+0.0i 1632.0+0.0i 1310.0+0.0i 1432.0+0.0i 872.0+0.0i 1024.0+0.0i 822.0+0.0i 944.0+0.0i 264.0+0.0i 416.0+0.0i 334.0+0.0i 456.0+0.0i 880.0+0.0i 972.0+0.0i 1792.0+0.0i 1976.0+0.0i 512.0+0.0i 604.0+0.0i 1056.0+0.0i 1240.0+0.0i 144.0+0.0i 236.0+0.0i 320.0+0.0i 504.0+0.0i 1588.0+0.0i 1736.0+0.0i 1072.0+0.0i 1184.0+0.0i 996.0+0.0i 1144.0+0.0i 624.0+0.0i 736.0+0.0i 404.0+0.0i 552.0+0.0i 176.0+0.0i 288.0+0.0i]]
LinearWrapper_1.vector
[[0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i]]
